ecoscope/plotting/plot.py

```diff
@@ -386,3 +386,98 @@ def pie_chart(
+
+
+def draw_historic_timeseries(
+    df: pd.DataFrame,
+    current_value_column: str,
+    current_value_title: str,
+    historic_min_column: str = None,
+    historic_max_column: str = None,
+    historic_band_title: str = "Historic Min-Max",
+    historic_mean_column: str = None,
+    historic_mean_title: str = "Historic Mean",
+    layout_kwargs: dict = None,
+):
+    """
+    Creates a timeseries plot compared with historical values
+    Parameters
+    ----------
+    df: pd.Dataframe
+        The data to plot
+    current_value_column: str
+        The name of the dataframe column to pull slice values from
+    current_value_title: str
+        The title shown in the plot legend for current value
+    historic_min_column: str
+        The name of the dataframe column to pull historic min values from. historic_min_column and historic_max_column should exist together.
+    historic_max_column: str
+        The name of the dataframe column to pull historic max values from. historic_min_column and historic_max_column should exist together.
+    historic_band_title: str
+        The title shown in the plot legend for historic band
+    historic_mean_column: str
+        The name of the dataframe column to pull historic mean values from
+    current_value_title: str
+        The title shown in the plot legend for historic mean values
+    layout_kwargs: dict
+        Additional kwargs passed to plotly.go.Figure(layout)
+    Returns
+    -------
+    fig : plotly.graph_objects.Figure
+        The plotly bar chart
+    """
+
+    fig = go.Figure(layout=layout_kwargs)
+
+    if historic_max_column and historic_min_column:
+        # add the upper bound
+        fig.add_trace(
+            go.Scatter(
+                x=df.img_date,
+                y=df[historic_max_column],
+                fill=None,
+                mode="lines",
+                line_color="green",
+                name="",
+                showlegend=False,
+            )
+        )
+        
+        # lower band
+        fig.add_trace(
+            go.Scatter(
+                x=df.img_date,
+                y=df[historic_min_column],
+                fill="tonexty",
+                mode="lines",
+                line_color="green",
+                name=historic_band_title,
+            )
+        )
+
+    if historic_mean_column:
+        # add the historic mean
+        fig.add_trace(
+            go.Scatter(
+                x=df.img_date,
+                y=df[historic_mean_column],
+                fill=None,
+                mode="lines",
+                line=dict(color="green", dash="dot"),
+                name=historic_mean_title,
+            )
+        )
+
+    # add current NDVI values
+    fig.add_trace(
+        go.Scatter(
+            x=df.img_date,
+            y=df[current_value_column],
+            fill=None,
+            mode="lines",
+            line_color="navy",
+            name=current_value_title,
+        )
+    )
+
+    return fig
```

Comment:
Would it make sense if we had a set of style kwargs for the upper/lower bands, the historic mean and the current value? 

(I'm including `mode` because I think allowing a choice between "lines", "markers" could be useful)

Ie;
```python
def draw_historic_timeseries(
    df: pd.DataFrame,
    .
    .
    upper_lower_band_style: dict = {mode:"lines", line_color:"green"},
    historic_mean_style: dict = {mode:"lines", "line": {"color":"green", "dash":"dot")},
    current_value_style: dict = {mode:"lines", line_color:"navy"},
    layout_kwargs: dict = None,
):
    .
    .
        # add the upper bound
        fig.add_trace(
            go.Scatter(
                x=df.img_date,
                y=df[historic_max_column],
                fill=None,
                name="",
                showlegend=False,
                **upper_lower_band_style,
            )
        )
        
        # lower band
        fig.add_trace(
            go.Scatter(
                x=df.img_date,
                y=df[historic_min_column],
                fill="tonexty",
                name=historic_band_title,
                **upper_lower_band_style,
            )
        )

    if historic_mean_column:
        # add the historic mean
        fig.add_trace(
            go.Scatter(
                x=df.img_date,
                y=df[historic_mean_column],
                fill=None,
                name=historic_mean_title,
                **historic_mean_style,
            )
        )

    # add current NDVI values
    fig.add_trace(
        go.Scatter(
            x=df.img_date,
            y=df[current_value_column],
            fill=None,
            name=current_value_title,
            **current_value_style,
        )
    )

    return fig
```
Sorry I tried to do this as a suggestion but the diff was illegible :sweat:

Reply:
Thanks for the great advise! Added them inline.